### ISSUE TITLE
Discord chat plugin

### ISSUE THREAD

**Issue by chr1st0ph3rGG** [wasurenakusa/engine#10]
Add a plugin so communication can happen via discord

- [x] Receive dms from a user in discord
- [x] Be able to send messages to the user
- [ ] #30

**Comment by chr1st0ph3rGG** [comment]
Hmmm im thinking about how we could provide the functionality for other addons to create chat commands…  one possibility is that in our base class we could add a add_commands method that the discord bot call before starting and dynamically generating the bot commands… the structure of how the command structure looks like should be not specific to discord and should simply provide a callback function that would run … 

regarding python discord frameworks i think I will use interactions.py as it provides the command functionality… ( I looked into discord.py but i dindt figure out how to add real slash commands etc (beside that hybrid stuff)

**Comment by chr1st0ph3rGG** [comment]
The remaining task is now in its own issue
